R/parse.R

```diff
@@ -27,14 +27,19 @@ make_timestamp <- function(input) {
   safe_format(input, "%Y-%m-%dT%H:%M:%SZ", tz = "UTC", usetz = FALSE)
 }
 
-ensure_columns <- function(.data, ptype) {
+ensure_columns <- function(.data, ptype, order_columns = FALSE) {
   # Given a prototype, ensure that all columns are present and cast to the correct type.
   # If a column is missing in .data, it will be created with all missing values of the correct type.
   # If a column is present in both, it will be cast to the correct type.
   # If a column is present in .data but not in ptype, it will be left as is.
   for (i in names(ptype)) {
     .data <- ensure_column(.data, ptype[[i]], i)
   }
+
+  if (order_columns) {
+    .data <- .data[, names(ptype), drop = FALSE]
```

Comment:
We should try to keep the names of data as well, in case it has more columns than ptype.

Untested!

```suggestion
    .data <- .data[, unique(c(names(ptype), names(.data))), drop = FALSE]
```

Reply:
In a newer pull request I changed this argument name to `use_ptype_cols` to make this part of the behavior more clear (that behavior is used to drop the `guid` column). But perhaps it's better to decompose those two behaviors (reordering, keeping only ptype columns).